library/src/main/scala/scala/scalajs/js/FinalizationRegistry.scala

```diff
@@ -0,0 +1,63 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package scala.scalajs.js
+
+import scala.scalajs.js
+import scala.scalajs.js.annotation.JSGlobal
+
+/** <span class="badge badge-ecma2021" style="float: right;">ECMAScript 2021</span>
+ *
+ *  A FinalizationRegistry object lets you request a callback when an object is garbage-collected.
+ *
+ *  FinalizationRegistry provides a way to request that a cleanup callback get called at some point
+ *  when an object registered with the registry has been reclaimed (garbage-collected).
+ *  (Cleanup callbacks are sometimes called finalizers.)
+ *
+ *  MDN
+ *
+ *  @tparam A Type of the target object to register
+ *  @tparam B Type of value to pass to the finalizer
+ *  @tparam C Type of an unregistration token
+ */
+@js.native
+@JSGlobal
+class FinalizationRegistry[A, B, C](finalizer: js.Function1[B, Unit]) extends js.Object {
```

Comment:
We should use contravariant type parameters. In addition, there is no need to constrain the result type of `finalizer` to be Unit:
```suggestion
class FinalizationRegistry[-A, -B, -C](finalizer: js.Function1[B, scala.Any]) extends js.Object {
```
Using contravariant type params also helps with type inference, as `A` and `C` will be inferred to be `Any` by default, rather than `Nothing`.